hasura/graphql-engine
# read/update migration settings from cli

### Description
This PR adds a `settings` subcommand to the `migration` command and contains two further subcommands `set` and `get`.

### Affected components 
- CLI

### Related Issues
#2290 

### Solution and Design
Mostly follows patterns of other CLI commands, nothing new in design here. Not sure if `SUCCESS` is the right thing to print when `set` runs successfully.

### Steps to test and verify
1. Spin up the test docker container
2. Build the cli and copy it into the container
  (`docker cp _output/cli-settings-de3bd50-dirty/cli-hasura-linux-amd64 a308:/bin`)
3. Init a migration folder `/bin/li-hasura-linux-amd64 init` and `cd` into it.
4. Run `cli-hasura-linux-amd64 migrate settings get migration_mode` should get "true"
5. `cli-hasura-linux-amd64 migrate settings set migration_mode=false` should print "SUCCESS"
6. Now `cli-hasura-linux-amd64 migrate settings get migration_mode` should return "false" and the actual console switch is off if you inspect it as well.


### Limitations, known bugs & workarounds
I wonder about tests. I didn't add any and I'm not sure I know enough about Go to really write them effectively. I noticed that you can set any string to any value here and if you try to get a value that doesn't exist you get an unhandled error, but those are in the underlying driver it seems.

fix issue #2290
